src/IIIFPresentation/API/Features/Manifest/CanvasPaintingResolver.cs

```diff
@@ -211,13 +223,24 @@ public class CanvasPaintingResolver(
                 StaticWidth = paintedResource.CanvasPainting?.StaticWidth,
                 StaticHeight = paintedResource.CanvasPainting?.StaticHeight
             };
+            
+            if (canvasPaintings.Where(c => c.CanvasOrder != cp.CanvasOrder).Any(c => c.Id == cp.Id))
+            {
+                return (ErrorHelper.DuplicateCanvasId<PresentationManifest>(specifiedCanvasId), null);
+            }
+
+            if (canvasPaintings.Where(c => c.CanvasOrder == cp.CanvasOrder).Any(c => c.Id != cp.Id))
+            {
+                return (ErrorHelper.CanvasOrderDifferentCanvasId<PresentationManifest>(paintedResource.CanvasPainting?.CanvasId),
+                    null);
+            };
 
             count++;
             canvasPaintings.Add(cp);
         }
         return (null, canvasPaintings);
 
-        string GetCanvasId(Models.API.Manifest.CanvasPainting? cp)
+        string GetGeneratedCanvasId(Models.API.Manifest.CanvasPainting? cp)
```

Comment:
Should the method of getting CanvasIds be unified? There's currently 2 different methods to getting canvasIds - `GetGeneratedCanvasId` which returns canvasId that we've minted and `GetCanvasId` which returns canvasId that we've been provided. 

Seems like it would be preferably to combine these methods into 1 that can return the appropriate canvasId regardless of whether it was provided or minted?